lib/gollum-lib/filter/emoji.rb

```diff
@@ -0,0 +1,23 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Emoji
+#
+# Render emoji such as :smile:
+class Gollum::Filter::Emoji < Gollum::Filter
+  EMOJI_TAG_PATTERN = %r{
+    (?<!\[{2})
+    :(?:([\w-]|</?em>)+):
+    (?!\]{^2})
+  }ix
+
+  def extract(data)
+    data
+  end
+
+  def process(data)
+    data.gsub EMOJI_TAG_PATTERN do |tag|
+      name = tag[1..-2].gsub(%r{</?em>}, '_')   # undo emphasis
```

Comment:
Actually that is why the filter chain is traversed twice (the second time in reverse). You could extract `name` in the `extract` method and replace it with a temporary placeholder string that is not in danger of being parsed by another filter (the tags filter uses `"TAG#{Digest::SHA1.hexdigest(tag)}TAG"`). Then, on the second pass, you can replace that placeholder string with your HTML snippet in the `process` method (`name` should still be available). Take a look at the tags filter for an example.